draft-ietf-rtcweb-jsep.xml

```diff
@@ -3929,13 +3920,16 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
             that the JSEP implementation should be sending media
             ("sendonly" for local answers, "recvonly" for remote
             answers, or "sendrecv" for either type of answer),
-            choose the media format to send
-            as the most preferred media format from the remote
-            description that is also present in the answer, as
-            described in
-            <xref target="RFC3264" />, Sections 6.1 and 7, and start
-            transmitting RTP media once the underlying transport layers
-            have been established. If an SSRC has not already been
+            choose the media format to send; for the offerer, this will be
+            the most preferred media format from the answer that is also
+            locally supported, as described in <xref target="RFC3264" />,
+            Section 7; for the answerer, this will be
```

Comment:
> Does that outcome seem right? (Does the spec allow an app to cause that to happen?)

RFC3264 allows it to happen. But if Bob is a JSEP implementation, he can't answer with opus, because of additional restrictions JSEP adds.

> Of 3264 has the constraint that the answer can not send with a codec it is not capable of receiving.

... If it's a "sendrecv" m= section. If it's "sendonly" (in the answer) this restriction doesn't apply.

> I think the thing that is strange is that the offerer can send with a codec that wasn't in the offer, but the answerer cannot send with a codec that wasn't in the answer.

SDP is indeed strange.

> we could ensure uniformity and force the offerer to only send things it offered.

I don't see why we should add even more additional restrictions to JSEP. Plus we've been over this several times before (at IETF meetings and editor meetings) and agreed it should be legal.